File: web/src/request/modules/productLine.js
import { createAPI } from '../requestUtils';
import request from '@/plugin/axios';
const productLine = {
  // 设备的增删改查
  line (opts = {}) {
    console.log(opts,999);
    
    const { method = 'GET' } = opts
    const paramsKey = method === 'GET' ? 'params' : 'data'
    const {
      [paramsKey]: {
        id = false,
        name = '',
        image = '',
        File = '',
        level = '',
        location = '',
        life = '',
        installTime = ''
      } = {}
    } = opts
    console.log(name,
      image,
      File,
      level,
      location,
      life,
      installTime);
      let data = new FormData()
      data.append("name",name)
      data.append("image",image)
      data.append("File",File)
      data.append("level",level)
      data.append("location",location)
      data.append("life",life)
      data.append("installTime",installTime)
    // return createAPI(
    //   {
    //     url: `ProductLine${id ? '/' + id : ''}`,
    //     method,
    //     headers: {
    //       // 'Content-Type': 'application/x-www-form-urlencoded'
    //       'Content-Type': 'multipart/form-data',
    //       'test': 11
    //     },
    //     // [paramsKey]: {
    //     //   name,
    //     //   image,
    //     //   File,
    //     //   level,
    //     //   location,
    //     //   life,
    //     //   installTime,
    //     // }
    //   }
      return request({
        url: `ProductLine${id ? '/' + id : ''}`,
        method,
        headers: {
          // 'Content-Type': 'application/x-www-form-urlencoded'
          'Content-Type': 'multipart/form-data'
        },
        [paramsKey]: data
      }
    )
  },
  updateLine (opts = {}) {
    const { method = 'GET' } = opts
    const paramsKey = method === 'GET' ? 'params' : 'data'
    const {
      [paramsKey]: {
        id = '',
        name = '',
        File = '',
        level = '',
        location = '',
        life = '',
        installTime = ''
      } = {}
    } = opts
    let data = new FormData()
      data.append("name",name)
      data.append("id",id)
      data.append("File",File)
      data.append("level",level)
      data.append("location",location)
      data.append("life",life)
      data.append("installTime",installTime)
    return request(
      {
        url: `ProductLine`,
        method,
        headers: {
          'Content-Type': 'multipart/form-data'
        },
        [paramsKey]: data
        // [paramsKey]: {
        //   id,
        //   name,
        //   File,
        //   level,
        //   location,
        //   life,
        //   installTime,
        // }
      }
    )
  },
}
export default productLine

File: NBWebApp/Models/ProductLine.cs
namespace NBWebApp.Models
{
    /// <summary>
    /// 产线
    /// </summary>
    public class ProductLine
    {
        /// <summary>
        /// KEY
        /// </summary>
        public int Id { get; set; }
        /// <summary>
        /// 产线名称
        /// </summary>
        public string Name { get; set; } = "";
        /// <summary>
        /// 产线图片
        /// </summary>
        public string Image { get; set; }
        /// <summary>
        /// 检测等级
        /// </summary>
        public string Level { get; set; } = "";
        /// <summary>
        /// 位置
        /// </summary>
        public string Location { get; set; } = "";
        /// <summary>
        /// 设备寿命
        /// </summary>
        public string Life { get; set; } = "";
        /// <summary>
        /// 安装时间
        /// </summary>
        public DateTime? InstallTime { get; set; }
    }
    /// <summary>
    /// 产线保存
    /// </summary>
    public class ProductLineSave : ProductLine
    {
        /// <summary>
        /// 图片文件
        /// </summary>
        public IFormFile File { get; set; }
    }
}
